### ISSUE TITLE
Squirrel for Windows

### ISSUE THREAD

**Issue by iamdriz** [atom/electron#939]
Are you planning on bring Squirrel for Windows support into atom-shell?

We're looking to use atom-shell for an upcoming project but it needs an update framework and needs to run on Windows... (I wish we could just use OS X)

**Comment by iamdriz** [comment]
Now that atom shell has become Electron, has any progress been made to add this for Windows yet?

Thanks.

**Comment by paulcbetts** [comment]
Nope! Wanna work on it?

**Comment by hardchor** [comment]
The new http://electron.atom.io/ page lists windows installers?

**Comment by Devagamster** [comment]
Id like to help in any way I can. Is there an architecture that is already in mind, or is it all up in the air? Will the final api for windows installers try to emulate the api currently available for mac or will it be different? I have a simple electron app running at the moment, so I have a simple view of how it might work.

**Comment by patrickdet** [comment]
i just implemented auto updates on windows for a client project and it was quite a handful. @iamdriz if you need to set it up for a project, i hope you have some time. it helps to check how atom has implemented it. i think i would like to help out with a pull request to get the same nice api's that are available on the mac side but i lack the c++ skills to write those bindings

**Comment by paulcbetts** [comment]
> i think i would like to help out with a pull request to get the same nice api's that are available on the mac side but i lack the c++ skills to write those bindings

You would probably actually write these all in Coffeescript - unlike Squirrel.Mac, Electron-based apps interact with Squirrel.Windows via just running a process (Update.exe) and reading stdout/stderr and error codes

**Comment by iamdriz** [comment]
Unfortunately I have no skill set in this area to offer any help. I am more than happy to test things and report issues I find as I have been doing. Regarding the updater that atom editor uses... Can that not be packaged as part of electron? So move it down a level into the core framework.

**Comment by patrickdet** [comment]
@paulcbetts some guidance would be awesome! if it's just coffeescript then i could help out. what has been hard to debug is how clean the update goes. in my situation an updated version of the app gets installed but i have to quit the app manually and launch again. when i check the Squirrel update logs they say something like "waiting for PID to quit" or so. Also the cleanup does not happen, which means that after a few upgrades i end up with lots of squirrel temp files and old application versions. so if i can help to get a clean version of the updater into electron i'd be happy to help

**Comment by patrickdet** [comment]
I think another blocking issue for a unified api between the mac and windows autoUpdater is this https://github.com/Squirrel/Squirrel.Windows/issues/132 which happens when you set a feed url like this

```
autoUpdater.setFeedUrl('http://mycompany.com/myapp/latest?version=' + app.getVersion());
```

**Comment by paulcbetts** [comment]
> in my situation an updated version of the app gets installed but i have to quit the app manually and launch again.

This is by-design, Squirrel never will interrupt a user's work. You shouldn't either :)

> Also the cleanup does not happen, which means that after a few upgrades i end up with lots of squirrel temp files and old application versions.

Hm, you should only be seeing one extra version (i.e. two `app-` folders). What version of Squirrel are you building against?

> I think another blocking issue for a unified api between the mac and windows autoUpdater is this Squirrel/Squirrel.Windows#132 which happens when you set a feed url like this

You don't need to do this, Squirrel update "servers" are just S3 buckets / anything that serves up static HTML

**Comment by patrickdet** [comment]
i have put something together. not much but it's something to get the conversation going https://github.com/atom/electron/pull/1984

**Comment by zcbenz** [comment]
It is now included in v0.34.1.
